### PR TITLE
Move ast2400 platforms to the dev-4.6 kernel tree

### PR DESCRIPTION
**Pull request by shenki**
The defconfig is regnerated with the newer kernel base. A large number
of unused options are disabled to reduce the kernel binary size.

Signed-off-by: Joel Stanley <joel@jms.id.au>

---
This change is [Reviewable](https://reviewable.io/reviews/openbmc/openbmc/467)

### CODE REVIEW

File: meta-openbmc-bsp/meta-aspeed/meta-ast2400/recipes-kernel/linux/linux-obmc/defconfig [hunk 1]
```diff
@@ -56,132 +55,83 @@ CONFIG_MTD=y
...
-# CONFIG_SERIAL_8250_DMA is not set
 CONFIG_SERIAL_8250_NR_UARTS=6
 CONFIG_SERIAL_8250_RUNTIME_UARTS=6
 CONFIG_SERIAL_8250_EXTENDED=y
 CONFIG_SERIAL_8250_SHARE_IRQ=y
-CONFIG_SERIAL_ASPEED_VUART=y
 CONFIG_SERIAL_OF_PLATFORM=y
-CONFIG_HW_RANDOM=y
-CONFIG_I2C=y
+# CONFIG_HW_RANDOM is not set
```
Comment (williamspatrick) [inline meta-openbmc-bsp/meta-aspeed/meta-ast2400/recipes-kernel/linux/linux-obmc/defconfig]: Isn't there a HW RNG on the AST2400/AST2500?  We should use that if available I think.

File: meta-openbmc-bsp/meta-aspeed/meta-ast2400/recipes-kernel/linux/linux-obmc/defconfig [hunk 2]
```diff
@@ -56,132 +55,83 @@ CONFIG_MTD=y
...
 CONFIG_SERIAL_8250_CONSOLE=y
-# CONFIG_SERIAL_8250_DMA is not set
 CONFIG_SERIAL_8250_NR_UARTS=6
 CONFIG_SERIAL_8250_RUNTIME_UARTS=6
 CONFIG_SERIAL_8250_EXTENDED=y
 CONFIG_SERIAL_8250_SHARE_IRQ=y
-CONFIG_SERIAL_ASPEED_VUART=y
 CONFIG_SERIAL_OF_PLATFORM=y
-CONFIG_HW_RANDOM=y
-CONFIG_I2C=y
```
Comment (williamspatrick) [inline meta-openbmc-bsp/meta-aspeed/meta-ast2400/recipes-kernel/linux/linux-obmc/defconfig]: Why are the I2C configs removed?

Comment (shenki) [inline meta-openbmc-bsp/meta-aspeed/meta-ast2400/recipes-kernel/linux/linux-obmc/defconfig]: This is a defconfig, it contains the minimal set of `CONFIG_` options required to describe the options we want enabled in a system. `CONFIG_I2C` is `select`ed by one of the other parts of the build, so the tooling decides it's not required in the defconfig.

I think in this case selecting the ast2400 machine forces on the I2C driver and therefore the I2C subsystem.

File: meta-openbmc-bsp/meta-aspeed/meta-ast2400/recipes-kernel/linux/linux-obmc/defconfig (hunk 1 above)
Comment (shenki) [inline meta-openbmc-bsp/meta-aspeed/meta-ast2400/recipes-kernel/linux/linux-obmc/defconfig]: There is, however we do not have code for it yet. Patches to write a driver for it are welcome.

**Pull request merged**